lib/rouge/lexer.rb

```diff
@@ -23,6 +23,14 @@ def lex(stream, opts={}, &b)
         new(opts).lex(stream, &b)
       end
 
+      # In case #continue_lex is called statically, we simply
+      # begin a new lex from the beginning, since there is no state.
+      #
+      # @see #continue_lex
+      def continue_lex(*a, &b)
```

Comment:
@ashmaroli Was your comment about this line? Consumers shouldn't really be calling this method; I think @jneen has it here to handle edge cases.

Reply:
My comment was triggered by that line and several other uses of `%b` in the codebase..
However, I admit that it cannot be avoided in all situations..

Reply:
For Rubies >= 2.5, the block overhead [might be mostly avoided](https://blog.bigbinary.com/2018/05/22/ruby-2-5-added-lazy-proc-allocation-for-block-parameters.html).

Reply:
I think the true optimization was made only in Ruby 2.6.. Ruby 2.5 laid the foundation for it.. Not sure though..
Either ways, thanks for the link.

Reply:
Ruby 2.6 offered a speed up for code that uses `Proc#call` rather than `yield` (if I'm reading [this announcement](https://www.ruby-lang.org/en/news/2018/12/25/ruby-2-6-0-released/) correctly).

Reply:
In code for streaming values, this is unavoidable, since we need to pass around the output stream into different contexts. Luckily the allocation *should* only be done once for us by `enum_for`.

Reply:
Ah yes. Both `Proc#call` and `block.call` have been sped up in Ruby 2.6 :+1:

Reply:
(as for the array allocation, most uses should call the instance method `#continue_lex` directly and avoid that code path anyways)